src/Microsoft.Sbom.Api/Executors/Spdx3SerializationStrategy.cs

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.Sbom.Api.Entities;
+using Microsoft.Sbom.Api.Utils;
+using Microsoft.Sbom.Extensions;
+
+namespace Microsoft.Sbom.Api.Workflows.Helpers;
+
+/// <summary>
+/// Serialization methods for SPDX 3.0.
+/// </summary>
+public class Spdx3SerializationStrategy : IJsonSerializationStrategy
+{
+    public void AddToFilesSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config)
+    {
+        elementsSupportingConfigs.Add(config);
+    }
+
+    public void AddToPackagesSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config)
+    {
+        elementsSupportingConfigs.Add(config);
+    }
+
+    public bool AddToRelationshipsSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config)
+    {
+        return true;
+    }
+
+    public void AddToExternalDocRefsSupportingConfig(IList<ISbomConfig> elementsSupportingConfigs, ISbomConfig config)
+    {
+        elementsSupportingConfigs.Add(config);
+    }
+
+    public async Task<List<FileValidationResult>> WriteJsonObjectsToSbomAsync(
+        ISbomConfig sbomConfig,
+        string spdxManifestVersion,
+        IJsonArrayGenerator<FileArrayGenerator> fileArrayGenerator,
+        IJsonArrayGenerator<PackageArrayGenerator> packageArrayGenerator,
+        IJsonArrayGenerator<RelationshipsArrayGenerator> relationshipsArrayGenerator,
+        IJsonArrayGenerator<ExternalDocumentReferenceGenerator> externalDocumentReferenceGenerator)
+    {
+        fileArrayGenerator.SbomConfig = sbomConfig;
+        packageArrayGenerator.SbomConfig = sbomConfig;
+        relationshipsArrayGenerator.SbomConfig = sbomConfig;
+        externalDocumentReferenceGenerator.SbomConfig = sbomConfig;
+
+        fileArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        packageArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        relationshipsArrayGenerator.SpdxManifestVersion = spdxManifestVersion;
+        externalDocumentReferenceGenerator.SpdxManifestVersion = spdxManifestVersion;
+
+        // Holds the SPDX IDs of all the elements that have been written to the SBOM. Used for deduplication.
+        var elementsSpdxIdList = new HashSet<string>();
+
+        WriteContext(sbomConfig);
+
+        sbomConfig.JsonSerializer.StartJsonArray(Constants.SPDXGraphHeaderName);
+
+        // Files section
+        var generateResult = await fileArrayGenerator.GenerateAsync();
+        WriteElementsToSbom(generateResult, elementsSpdxIdList);
+
+        // Packages section
+        var packagesGenerateResult = await packageArrayGenerator.GenerateAsync();
```

Comment:
Variable name didn't change when `GenerateResult` type was renamed to `GenerationResult`